### PR TITLE
kvp reworks for read/writing

### PR DESCRIPTION
**Pull request by baude**
flush through to allow for kvp writing and reading.  in ignition, we may very well have to write the pool files to the rootfs.

### CODE REVIEW

**Comment by baude** [comment]
@n1hility ptal and merge

File: pkg/kvp/config.go [hunk 1]
```diff
@@ -66,6 +74,74 @@ type hvKvpMsgRet struct {
...
+	kv[poolID] = append(vps, vp)
+}
+
+func (kv KeyValuePair) WriteToFS(path string) error {
+	if err := os.MkdirAll(path, 777); err != nil {
+		return err
+	}
+	for poolID := range kv {
+		fqWritePath := filepath.Join(path, fmt.Sprintf("%s%d", DefaultKVPBaseName, poolID))
+		fmt.Println(fqWritePath)
```
Comment (n1hility) [inline pkg/kvp/config.go:130]: Leftover debugging?

File: pkg/kvp/kvp.go [hunk 2]
```diff
@@ -100,46 +105,44 @@ next:
...
 		counter++
 	}
 	if len(parts) < 1 {
-		return "", kvps, ErrNoKeyValuePairsFound
+		return "", nil, ErrNoKeyValuePairsFound
 	}
 	return strings.Join(parts, ""), leftOvers, nil
 }
+
+func makeEmptyFixedArray(len uint) []byte {
```
Comment (n1hility) [inline pkg/kvp/kvp.go:142]: I dont think you need this since make() zero inits slice

Comment (n1hility) [inline pkg/kvp/kvp.go:142]: I dont think you need this since make() zero inits slices

File: pkg/kvp/kvp.go [hunk 3]
```diff
@@ -64,25 +67,27 @@ next:
...
-		if l != sizeOf {
-			return nil, ErrUnableToReadFromKVP
-		}
 		if err != nil {
+			if err == unix.EAGAIN || err == unix.EWOULDBLOCK {
+				continue
+			}
 			return nil, err
 		}
+		if l != sizeOf {
```
Comment (n1hility) [inline pkg/kvp/kvp.go:76]: Do you know if this bus device guarantees non-partial reads? I'm assuming it does but might be worth comparing with the daemon code.

**Pull request merged**